test/strategies/douglas_peucker.cpp

```diff
@@ -221,12 +225,15 @@ inline void test_with_strategy()
 
     typedef bg::model::point<CoordinateType, 2, bg::cs::cartesian> point_type;
     typedef bg::model::linestring<point_type> linestring_type;
+    typedef bg::model::segment<point_type> segment_type;
     typedef test_one_case<linestring_type> tester;
 
     linestring_type ls1, ls2;
     linestring_type res1, res2;
+    point_type const p1(-6,-13), p2(0,-15);
+    segment_type const s(point_type(12,-3), point_type(-12,5));
 
-    if (boost::is_same<CoordinateType, long double>::value)
+    if (bg::comparable_distance(p1, s) >= bg::comparable_distance(p2, s))
     {
         tester::apply("LINESTRING(12 -3, 4 8,-6 -13,-9 4,0 -15,-12 5)",
                       10,
```

Comment:
With "I don't know the algorithm", I meant that I don't know which algorithm is used in SQL Server. Which was indeed not clear from my comment.

The Douglas Peucker algorithm I know very well, I implemented all this code, first together with a colluege back in 1995, and in 2007 again for Boost.Geometry...

For the rest, thanks for your explanations, did not yet read them all but will take some time for that tomorrow.